core/src/main/java/org/infinispan/statetransfer/StateConsumerImpl.java

```diff
@@ -680,6 +699,13 @@ protected IntSet getOwnedSegments(ConsistentHash consistentHash) {
       return CompletableFutures.completedNull();
    }
 
+   protected void onCompletedSegment(Integer segmentId) {
```

Comment:
```suggestion
   protected void onCompletedSegment(int segmentId) {
```
I am not 100% if the JVM will remove the auto box for scattered state, it may very well, but we should clean this up a bit.

To avoid the double boxking in this method we should use a separate Integer variable in this method and change the stopTrackFor method to take Integer.